File: codility/NumberSolitaire.js
function solution(A) {
    // Implement your solution here
    const arr = new Array(A.length);
    arr[0] = A[0];
    arr[1] = A[0] + A[1];
    for (let i = 2; i < arr.length; i++) {
        let max = arr[i - 1];
        if (arr[i - 2] !== undefined && arr[i - 2] > max) max = arr[i - 2]
        if (arr[i - 3] !== undefined && arr[i - 3] > max) max = arr[i - 3]
        if (arr[i - 4] !== undefined && arr[i - 4] > max) max = arr[i - 4]
        if (arr[i - 5] !== undefined && arr[i - 5] > max) max = arr[i - 5]
        if (arr[i - 6] !== undefined && arr[i - 6] > max) max = arr[i - 6]
        arr[i] = max + A[i];
    }
    return arr[A.length - 1];
}

File: codility/python/NumberSolitaire.py
def solution(A):
    # Implement your solution here
    n = len(A)
    result_list = [0] * n
    result_list[0] = A[0]
    result_list[1] = A[0]+A[1]
    for i in range(2, n):
        temp_list = [result_list[i-1], result_list[i-2]]
        if i > 2:
            temp_list.append(result_list[i-3])
        if i > 3:
            temp_list.append(result_list[i-4])
        if i > 4:
            temp_list.append(result_list[i-5])
        if i > 5:
            temp_list.append(result_list[i-6])
        result_list[i] = max(temp_list) + A[i]
    return result_list[-1]